encoding/jsonc_test.ts

```diff
@@ -171,3 +173,51 @@ Deno.test({
     assertValidParse('  ["foo\\\\\\\\", "bar"]', ["foo\\\\", "bar"]);
   },
 });
+
+Deno.test({
+  name: "[jsonc] avoid prototype pollution",
+  fn() {
+    // reference: https://github.com/advisories/GHSA-9c47-m6qq-7p4h
+    // Check for prototype pollution using the code contained in the CVE-2022-46175.
+
+    const jsonc = JSONC.parse('{"__proto__": { "isAdmin": true }}');
+    // @ts-expect-error: for test
+    assertEquals(jsonc.__proto__, { isAdmin: true });
+    // @ts-expect-error: for test
+    assertEquals(jsonc.isAdmin, undefined);
+    // @ts-expect-error: for test
+    assertEquals(Object.keys(jsonc), ["__proto__"]);
+  },
+});
+
+Deno.test({
+  name: "[jsonc] avoid prototype pollution in Node.js",
+  async fn() {
+    // reference: https://github.com/advisories/GHSA-9c47-m6qq-7p4h (CVE-2022-46175)
+    // Check whether prototype pollution does not occur when running in Node.js or browser environment using the code contained in the CVE-2022-46175.
+    const tempTile = await Deno.makeTempFile();
+    try {
+      // 1. generate code
+      const { code } = await bundle(
+        new URL(import.meta.resolve("./testdata/jsonc/node/check_proto.ts")),
+        { type: "classic" },
+      );
+
+      // 2. create tmp file
+      await Deno.writeTextFile(tempTile, code);
+
+      // 3. run in Node.js
+      const command = new Deno.Command("node", {
+        stdout: "inherit",
+        stderr: "inherit",
+        args: [tempTile],
+      });
```

Comment:
I added test in Node.js environment. However, if you don't want to have a dependency on Node.js, I can remove this test.

Reply:
I have updated the code so there is no longer a dependency on Node.js.